modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -573,6 +578,23 @@ public void endStructDef(String identifier) {
         this.varListStack.pop().forEach(structNode::addField);
         this.compUnit.addTopLevelNode(structNode);
     }
+
+    public void startEnumDef() {
+        this.enumStack.add(TreeBuilder.createEnumNode());
+    }
+
+    public void endEnumDef(String identifier) {
```

Comment:
Please add the diagnostic position to the enum node :)

Reply:
+1